src/client/chat/systems.rs

```diff
@@ -0,0 +1,288 @@
+use bevy::input::ButtonState;
+use chat_events::{FocusChangeEvent, FocusState, SendMessageEvent};
+
+use crate::prelude::*;
+
+const COLOR_UNFOCUSED: Color = Color::rgba(0.0, 0.0, 0.0, 0.0);
+const COLOR_FOCUSED: Color = Color::rgba(0.0, 0.0, 0.0, 0.5);
+
+pub fn setup_chat_container(mut commands: Commands) {
+    commands
+        .spawn(NodeBundle {
+            style: Style {
+                margin: UiRect::all(Val::Px(5.0)),
+                width: Val::Percent(50.0),
+                height: Val::Percent(80.0),
+                flex_direction: FlexDirection::ColumnReverse,
+                ..default()
+            },
+            background_color: BackgroundColor(COLOR_UNFOCUSED),
+            ..default()
+        })
+        .with_children(|parent| {
+            parent
+                .spawn(TextBundle {
+                    text: Text {
+                        sections: vec![TextSection {
+                            value: String::new(),
+                            style: TextStyle {
+                                font_size: 20.0,
+                                color: Color::rgba(1.0, 1.0, 1.0, 0.1),
```

Comment:
I would define common styles and colors in constants or small functions so you can reuse them.

Reply:
That's a reasonable idea.